src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.ts

```diff
@@ -46,21 +49,25 @@ export class ProgrammingAssessmentRepoExportDialogComponent implements OnInit {
             normalizeCodeStyle: false, // disabled by default because it is rather unstable
             hideStudentNameInZippedFolder: false,
         };
-        this.exerciseService
-            .find(this.exerciseId)
-            .pipe(
-                tap(({ body: exercise }) => {
-                    this.exercise = exercise!;
-                }),
-                catchError((err) => {
-                    this.jhiAlertService.error(err);
-                    this.clear();
-                    return of(undefined);
-                }),
-            )
-            .subscribe(() => {
-                this.isLoading = false;
-            });
+        if (this.selectedProgrammingExercises.length <= 0) {
+            this.exerciseService
+                .find(this.exerciseId)
+                .pipe(
+                    tap(({ body: exercise }) => {
+                        this.exercise = exercise!;
+                    }),
+                    catchError((err) => {
+                        this.jhiAlertService.error(err);
+                        this.clear();
+                        return of(null);
```

Comment:
I think `null` should be replaced with `undefined`

Reply:
You're right.. Fixed with 6885a6a :+1: